### PR TITLE
Avoid having module documentation links to itself

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

lots of modules

##### ANSIBLE VERSION

v2.3

##### SUMMARY

A lot of modules use M(own_module) in their documentation causing a link
in the documentation to itself. This cleans it up.

### CODE REVIEW

**Review by xen0l — APPROVED** [review]

File: lib/ansible/modules/commands/shell.py [hunk 1]
```diff
@@ -69,7 +69,7 @@
 notes:
    -  If you want to execute a command securely and predictably, it may be
       better to use the M(command) module instead. Best practices when writing
-      playbooks will follow the trend of using M(command) unless M(shell) is
+      playbooks will follow the trend of using M(command) unless C(shell) is
```
Comment (gundalow) [inline lib/ansible/modules/commands/shell.py]: I think this may be confusing to a reader now. They should be the same

Comment (dagwieers) [reply lib/ansible/modules/commands/shell.py]: Why ? If you are reading the shell documentation, you don't expect `shell` to be a link ? But the `command` documentation is quite relevant here.

Comment (dagwieers) [reply lib/ansible/modules/commands/shell.py]: Looking at the actual [produced documentation](http://docs.ansible.com/ansible/shell_module.html#notes), there is another issue. Links are not very clear from the note styles used. Everything appears in white on blue, even the links. However, I would prefer if `M( )` and `C( )` would also cause the text to be in proportional font so it is clear it is a module (or at least something special, and not just a word. If that would be the case, both `command` and `shell` would stand out, and only `command` would be a link.

Comment (gundalow) [reply lib/ansible/modules/commands/shell.py]: How about: "playbooks will follow the trend of using M(command) unless the C(shell) modules is"

Comment (dagwieers) [reply lib/ansible/modules/commands/shell.py]: Oh ok, that is indeed more clear. I will change this PR accordingly.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged, thanks :)

**Pull request merged**